### DISCUSSION TITLE
Find methods with specific annotation.

### DISCUSSION THREAD

**Discussion by WolandPL** [detekt/detekt#4552] [Q&A] [answered]
I'm trying to find all the methods that are annotated with Retrofit2 annotations (GET, PUT, etc).

I'm using `visitNamedFunction `and then `function.annotationEntries` which works fine, but although it's easy to get a simple name of the annotation, like "GET", I failed to get fully qualified name (to make sure that this is really a Retrofit annotation).

How can I get it?

**Comment by cortinico** [comment, marked as answer]
I would follow the same logic we have in the `AnnotationSuppressor` here:

https://github.com/detekt/detekt/blob/781395c748af45592d3a90993ed8d6a5808b0b8a/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/suppressors/AnnotationSuppressor.kt#L43-L52

Specifically if you're running with Type Resolution you can access to: `.fqNameOrNull(bindingContext)?.toString()` to get the fully qualified name of the annotation

**Comment by WolandPL** [comment]
Thanks for the answer @cortinico 

I'm trying to enable type resolution in my detektAll task definition but have problems with the classpath:
`Cannot set the value of read-only property 'classpath' for task ':detektAll' of type io.gitlab.arturbosch.detekt.Detekt.`

My task definition is:
```
tasks.register("detektAll", Detekt) {
    def autoFix = project.hasProperty('detektAutoFix')

    description = "Custom DETEKT build for all modules"
    parallel = true
    ignoreFailures = false
    autoCorrect = autoFix
    buildUponDefaultConfig = true
    setSource(projectSource)
    baseline.set(baselineFile)
    config.setFrom(configFile)
    include(kotlinFiles)
    exclude(resourceFiles, buildFiles)
    reports {
        html.enabled = true
        xml.enabled = false
        txt.enabled = false
    }
    jvmTarget = "1.8"
    classpath = buildFiles
}
```

**Reply by cortinico** [reply]
Is there a reason why you're not using the Detekt Gradle Plugin?

**Reply by WolandPL** [reply]
> Is there a reason why you're not using the Detekt Gradle Plugin?

I'd love to :) but is there a way to have a general task that would run e.g. detectVariant on all modules of the Android project? I only succeed to have them created separately for every module.

**Reply by cortinico** [reply]
If you invoke `./gradlew detektMain` that should do exactly what you're looking for

**Reply by WolandPL** [reply]
You're right, thanks!
